### PR TITLE
Add AWS Sigv4 Request Signer Sample Code

### PR DESCRIPTION
**Pull request by hanopensearch**
Signed-off-by: Han Lin <mlinhan@amazon.com>

### Description
Add AWS Sigv4 Request Signer sample code to the users guide.

### Issues Resolved
https://github.com/opensearch-project/opensearch-dsl-py/issues/58

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Yury-Fridlyand** [comment]
Is it possible to specify `access_key` and `secret_key` as function arguments? Are keys read from environment or `~/.aws/credentials` if they are not given?

**Comment by hanopensearch** [comment]
> Is it possible to specify `access_key` and `secret_key` as function arguments? Are keys read from environment or `~/.aws/credentials` if they are not given?

Yes keys are read from default aws profile. This is how to specify keys as function arguments:
```
# create credential from access_key and secret_key
credentials = botocore.credentials.Credentials(
    access_key=os.getenv('access_key'),
    secret_key=os.getenv('secret_key')
)
```

**Comment by Yury-Fridlyand** [comment]
> Yes keys are read from default aws profile. This is how to specify keys as function arguments:

Wonderful! Could you please add this to your examples? Thanks!

**Review by Yury-Fridlyand — APPROVED** [review]

File: USER_GUIDE.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 - [User Guide](#user-guide)
   - [Setup](#setup)
   - [Sample code](#sample-code)
+  - [AWS Sigv4 Request Signer Sample Code](#aws-sigv4-request-signer-sample-code)
```
Comment (dblock) [inline USER_GUIDE.md:4]: Maybe split this section into multiple samples under sample code? Sample Code > Creating an Index, and Sample Code -> AWS Sigv4

Comment (hanopensearch) [reply USER_GUIDE.md]: Updated the PR with the suggested change.

**Comment by Yury-Fridlyand** [comment]
@hanopensearch,
try to update your GitHub profile by making your email public - this will fix [DCO check failure](https://github.com/opensearch-project/opensearch-dsl-py/pull/65/checks?check_run_id=7739854718) for future commits.
Thanks!

**Comment by hanopensearch** [comment]
> @hanopensearch, try to update your GitHub profile by making your email public - this will fix [DCO check failure](https://github.com/opensearch-project/opensearch-dsl-py/pull/65/checks?check_run_id=7739854718) for future commits. Thanks!

Thanks for the tip. I made my email public. Do you know why integration tests failed? I only made change in the users guide.

**Comment by dblock** [comment]
@hanopensearch rebase this PR on top of https://github.com/opensearch-project/opensearch-dsl-py/pull/66, should pass tests

File: USER_GUIDE.md [hunk 2]
```diff
@@ -98,4 +102,81 @@ from opensearchpy import OpenSearch
...
+
+```
+from opensearch_dsl import Search
+from opensearchpy import OpenSearch, RequestsHttpConnection, AWSV4SignerAuth
+import botocore
+import boto3
+import os
+
+# cluster endpoint, for example: my-test-domain.us-east-1.es.amazonaws.com
+host = os.getenv('host')
```
Comment (dblock) [inline USER_GUIDE.md:117]: For an example, hardcode the domain instead of getting it from env

File: USER_GUIDE.md [hunk 3]
```diff
@@ -98,4 +102,81 @@ from opensearchpy import OpenSearch
...
+region = 'us-west-2' # e.g. us-west-1
```
Comment (dblock) [inline USER_GUIDE.md:118]: just hard-code `us-west-2`, the comment should be similar looking to host explaining what this is (credentials region scope)

File: USER_GUIDE.md [hunk 4]
```diff
@@ -98,4 +102,81 @@ from opensearchpy import OpenSearch
...
+
+# create credential from access_key and secret_key
+credentials = botocore.credentials.Credentials(
+    access_key=os.getenv('access_key'),
+    secret_key=os.getenv('secret_key')
```
Comment (dblock) [inline USER_GUIDE.md:123]: can I supply a session token too?

File: USER_GUIDE.md [hunk 5]
```diff
@@ -98,4 +102,81 @@ from opensearchpy import OpenSearch
...
+    http_auth = auth,
+    use_ssl = True,
+    verify_certs = True,
+    connection_class = RequestsHttpConnection
+)
+
+index_name = 'my-dsl-index'
+
+response = client.indices.create(index_name)
+print('\nCreating index:')
```
Comment (dblock) [inline USER_GUIDE.md:139]: Should be above the `response = ...`

File: USER_GUIDE.md [hunk 6]
```diff
@@ -98,4 +102,81 @@ from opensearchpy import OpenSearch
...
+id = '1'
+
+response = client.index(
+    index = index_name,
+    body = document,
+    id = id,
+    refresh = True
+)
+
+print('\nAdding document:')
```
Comment (dblock) [inline USER_GUIDE.md:157]: Same as above.

File: USER_GUIDE.md [hunk 7]
```diff
@@ -98,4 +102,81 @@ from opensearchpy import OpenSearch
...
+print(response)
+
+# Search for the document.
```
Comment (dblock) [inline USER_GUIDE.md:160]: Will this always work right away? In my experience there needs to be a little delay before search starts returning the document.

**Review by dblock — COMMENTED** [review]
I have some nitpicky suggestions :)

**Comment by VachaShah** [comment]
> > @hanopensearch, try to update your GitHub profile by making your email public - this will fix [DCO check failure](https://github.com/opensearch-project/opensearch-dsl-py/pull/65/checks?check_run_id=7739854718) for future commits. Thanks!
> 
> Thanks for the tip. I made my email public. Do you know why integration tests failed? I only made change in the users guide.

@hanopensearch It is failing because of #68

**Comment by VachaShah** [comment]
> > > @hanopensearch, try to update your GitHub profile by making your email public - this will fix [DCO check failure](https://github.com/opensearch-project/opensearch-dsl-py/pull/65/checks?check_run_id=7739854718) for future commits. Thanks!
> > 
> > 
> > Thanks for the tip. I made my email public. Do you know why integration tests failed? I only made change in the users guide.
> 
> @hanopensearch It is failing because of #68

Raised PR #69 with a fix.

**Comment by dblock** [comment]
@hanopensearch Want to finish this?

**Comment by wbeckler** [comment]
Given that we're migrating this library into the opensearch-py library, can we start over there? I'm closing this.

**Pull request closed without merging**